Firmware/RTK_Surveyor/System.ino

```diff
@@ -491,22 +612,24 @@ bool setMessagesUSB()
 {
   bool response = true;
 
-  response &= theGNSS.newCfgValset();
-  for (int x = 0 ; x < 36 ; x++)
+  int x = 0;
+  while (x < MAX_UBX_MSG)
   {
-    if (settings.ubxMessages[x].supported & zedModuleType)
-      response &= theGNSS.addCfgValset(settings.ubxMessages[x].msgConfigKey + 2, settings.ubxMessages[x].msgRate);
+    response &= theGNSS.newCfgValset();
+    
+    do
+    {
+      if (settings.ubxMessages[x].supported & zedModuleType)
+        response &= theGNSS.addCfgValset(settings.ubxMessages[x].msgConfigKey + 2, settings.ubxMessages[x].msgRate);
+      x++;
+    }
+    while (((x % 43) < 42) && (x < MAX_UBX_MSG)); // Limit 1st batch to 42. Batches after that will be (up to) 43 in size. It's a HHGTTG thing.
```

Comment:
Funny.

Reply:
Could have gone with 37 (73 / 2, rounded up). But what if we add a couple of extra messages...? You're then sending three blocks when two would do. The politically-correct answer is "use 63" (i.e. 1 less than 64). But the Marvin in me made me choose 42...